lib/importer/nunjucks/importer/macros/sheet_selector.njk

```diff
@@ -1,53 +1,114 @@
-    
-    {% if error %}
-    <p id="upload-error" class="govuk-error-message">
-      <span class="govuk-visually-hidden">Error:</span> {{ error.text }}
-    </p>
-    {% endif %}
-    
-    <div class="govuk-radios" data-module="govuk-radios">
+  <style>
+  .rd-sheet-preview {
+    width: 100%;
+    display: grid;
+    grid-template-columns: 44px 1fr;
+  }
+
+  .rd-sheet-preview:not(input,label,div.selectable) {
+    display: contents;
+  }
+
+  div.selectable { display: none; overflow-x: scroll}
+
+  input[type="radio"]:checked ~ div.selectable {
+    grid-row: 4 / -1;
+    display: block;
+  }
+
+  #previews {
+    margin-top: 2em;
+  }
+  </style>
+
+  <div class="govuk-form {% if error %}govuk-form-group--error{% endif %}">
+    <div class="govuk-form-group ">
+      {% if legend %}
+      <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+        <h1 class="govuk-fieldset__heading">
+          {{ legend }}
+        </h1>
+      </legend>
+      {% endif %}
+
+      {% if error %}
+      <p id="upload-error" class="govuk-error-message">
+        <span class="govuk-visually-hidden">Error:</span> {{ error.text }}
+      </p>
+      {% endif %}
+
+      <div class="govuk-radios rd-sheet-preview" data-module="govuk-radios">
         {% for sheet in sheets %}
         <div class="govuk-radios__item">
-            <input class="govuk-radios__input" id="{{sheet}}" name="sheet" type="radio" value="{{ sheet }}"  {% if
-      selectedSheet==sheet %}checked{% endif %}>
-        <label class="govuk-label govuk-radios__label" for="{{sheet}}">
-          {{sheet}}
-        </label>
-      </div>
+          <input class="govuk-radios__input" id="{{sheet.name}}" name="sheet" type="radio" value="{{sheet.name}}" {% if selectedSheet==sheet.name %}checked="checked"{% endif %} data-preview-index="{{loop.index0}}" onclick="javascript:preview(this);">
+          <label class="govuk-label govuk-radios__label" for="{{sheet.name}}">
+          {{sheet.name}}
+          </label>
+        </div> <!-- .govuk-radios__item  -->
         {% endfor %}
+      </div> <!-- .govuk-radios  -->
+  </div>
+</div>
+
+<div id="previews">
+  {% for sheet in sheets %}
+    <div class="selectable">
+      {% if sheet.data.rows == null %}
+      <div class="govuk-body">
+        Sheet '{{ sheet.name }}' is empty
+        </div>
+      {% else %}
+        <div  class="govuk-hint">
+          First {{sheet.data.rows | length}} rows of '{{sheet.name}}'
+        </div>
+        {{ importerTableView(sheet.data, hideHeader=true) }}
+      {% endif %}
     </div>
-  </fieldset>
+  {% endfor %}
 </div>
+
+<script>
+const previews = Array.from(document.getElementById("previews").children);
+
+// Preview any default selected options
+const currentlySelected = document.querySelector('input[type="radio"]:checked');
+if (currentlySelected && currentlySelected.dataset.previewIndex) {
+  previews[parseInt(currentlySelected.dataset.previewIndex)].style.display = "block";
+}
```

Comment:
Why not just call `preview(currentlySelected)`?

Reply:
Doh. Very good question. Fixed in ba64b75